File: infrastructure/db-core/src/main/java/com/nmh/commerce/utils/QuantityConverter.java
package com.nmh.commerce.utils;

import com.nmh.commerce.domain.Quantity;
import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

@Converter(autoApply = true)
public class QuantityConverter implements AttributeConverter<Quantity, Integer> {

    @Override
    public Integer convertToDatabaseColumn(Quantity quantity) {
        if (quantity == null) {
            return null;
        }
        return quantity.value;
    }

    @Override
    public Quantity convertToEntityAttribute(Integer integer) {
        if (integer == null) {
            return null;
        }
        return Quantity.of(integer);
    }
}

File: infrastructure/db-core/src/main/kotlin/com/nmh/commerce/utils/QuantityConverter.kt
package com.nmh.commerce.utils

import com.nmh.commerce.domain.Quantity
import com.nmh.commerce.domain.Quantity.Companion.of
import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter(autoApply = true)
class QuantityConverter : AttributeConverter<Quantity?, Int?> {
    override fun convertToDatabaseColumn(quantity: Quantity?): Int? {
        if (quantity == null) {
            return null
        }
        return quantity.value
    }

    override fun convertToEntityAttribute(integer: Int?): Quantity? {
        if (integer == null) {
            return null
        }
        return of(integer)
    }
}
